### ISSUE TITLE
Possible radiation calculation errors when used in a compact machine

### ISSUE THREAD

**Issue by viveleroi** [turbodiesel4598/NuclearCraft#560]
I tried putting a reactor inside a compact machine, thinking it would help avoid radiation problems near my base.

However, I ran a few TBU Oxide fuels in 3x3 reactor, pumped the depleted fuels out into my ME system, but when I travel to the compact machine I'm dead within seconds, even with heavy shielding on my armor. I took some rad-x and went back and the geiger counter showed the area was 410mRads, then 378, then 216, 99, and kept dropping.

Is there an issue with it being in a compact machine? How is TBU Oxide producing so much radiation? Even with rad-x I can't stay for long, I can't even imagine what higher fuels will do.

**Comment by turbodiesel4598** [comment]
When you say seconds, I’m guessing you mean a minute or so, as 400 mRad/t should not be near insta-killing you. If so there’s something else going on. Could you try not wearing shielded armour and looking at what the radiation level gets to (in creative, the radiation won’t kill you) while the reactor is running? 400 mRad/t with maximum shielding is roughly equivalent to about 1.5 Rad/t.

**Comment by viveleroi** [comment]
That reading was from 30 minutes later. I didn't spawn at my bed and lost everything so I had to get home.

No it was definitely seconds, roughly 5, guarantee it was less than 10. I entered the room, was poisoned, was dead before I heard five "attacks" and was wearing diamond armor with either medium or heavy shielding. Happened twice.

Unfortunately I took the reactor down because I needed to run it so testing it again won't be as easy. 

Here are steps I believe I took before this happened to me twice:

- Use compact machines with 1.12
- Make a large compact machine
- Build a 3x3 reactor for TBU Oxide. I can try to send the build somehow.
- Run 3 pellets while back in the overworld. Leave the depleted fuel in the fission controller overnight.
- Come back

**Comment by turbodiesel4598** [comment]
I’ll test this when I can, though I do have to say I’m skeptical :P If the radiation level really was 400 mRad/t and you really did die in seconds, I can’t think how it could have been the radiation that killed you.

Also, which version of NC are you using? Are the configs modified?

**Comment by viveleroi** [comment]
Keep in mind I said the levels were dropping, and the 400 was the first value I saw when I could return 30 minutes later. Who knows what it was when I first died. I was full health, this has happened twice, and it hasn't even come close after running the exact same reactor for ten times as long in the overworld after this.

NuclearCraft-2.17a-1.12.2

No config changes.

**Comment by turbodiesel4598** [comment]
And so I apologise for my skepticism! Actually found the bug causing this in another scenario, but this compact machine issue is basically the same thing. Fixed for the next version; thanks for the info :)

**Comment by viveleroi** [comment]
No worries, glad you figured it out. After I moved to my reactors back to the overworld (and far from my base) I never had any issues so I just stuck with it.

**Comment by viveleroi** [comment]
I assume this fix went out with the 2.17b release? It mentions a radiation bug when no neighbor chunks are loaded. I'm testing it now but it's early so I can't tell yet.

I did run some TBU in a small reactor in a compact machine for about 15 minutes. When I teleported in, the geiger showed 870 rads/t and quickly dropped down about 590 (it hovered around 600 or so in the overworld) (not using *any* shielding, etc).

I don't know why would it start high and fall but it makes me worried to check back in an hour.
